### DISCUSSION TITLE
Wan2.2 How to use Prodigy？

### DISCUSSION THREAD

**Discussion by YGG-Game** [kohya-ss/musubi-tuner#514] [Q&A]
I tried to use Prodigy to train Wan2.2, but it gave an error after one round of training.I printed the log and after one round of training, the step became 0.1.

Also, I'm using the Windows platform.

Parameter：
.\python\python.exe -m accelerate.commands.launch --num_cpu_threads_per_process 1 src/musubi_tuner/wan_train_network.py --task i2v-A14B ^
--dit models/stable_video_diffusion/wan2.2_i2v_high_noise_14B_fp16.safetensors ^
--vae models/vae/Wan2.1_VAE.pth ^
--t5 models/text_encoders/models_t5_umt5-xxl-enc-bf16.pth ^
--network_args loraplus_lr_ratio=4 ^
--dataset_config dataset/dataset_horse.toml ^
--flash_attn --split_attn ^
--mixed_precision fp16 --fp8_base --fp8_scaled ^
--optimizer_type prodigyopt.Prodigy ^
--learning_rate 1 ^
--optimizer_args weight_decay=0.01 betas=(0.9,0.99) ^
--lr_scheduler cosine_with_restarts --lr_scheduler_num_cycles 3 --lr_warmup_steps 100 ^
--gradient_checkpointing --gradient_accumulation_steps 1 ^
--max_data_loader_n_workers 8 --persistent_data_loader_workers ^
--network_module networks.lora_wan ^
--network_dim 16 --network_alpha 16 ^
--network_dropout 0.05 ^
--timestep_sampling shift --discrete_flow_shift 3.0 ^
--max_train_epochs 100 --save_every_n_epochs 10 ^
--seed 5 ^
--output_dir output ^
--output_name WAN2.2-HighNoise_Horse ^
--preserve_distribution_shape ^
--min_timestep 900 --max_timestep 1000 ^
--blocks_to_swap 30 ^
--log_with tensorboard ^
--logging_dir output/wan_logs ^
--save_state --save_state_on_train_end 

Error:
File "I:\AI\musubi-tuner\python\Lib\site-packages\prodigyopt\prodigy.py", line 152, in step raise RuntimeError(f"Setting different lr values in different parameter groups is only supported for values of 0")

**Comment by niceguy4** [comment]
Whaaaaaaaaaaaaaaaaaaaaaaaaaaaaaat?!?!? You can use Prodigy! That's awesome.

**Reply by YGG-Game** [reply]
After many attempts, I switched to using diffusion-pipe and it worked fine.

**Reply by kohya-ss** [reply]
> Whaaaaaaaaaaaaaaaaaaaaaaaaaaaaaat?!?!? You can use Prodigy! That's awesome.

After installing Prodigy with `pip install prodigyopt` , `--optimizer_type prodigyopt.Prodigy --learning_rate 1` options will work.

Please see [the official page](https://github.com/konstmish/prodigy?tab=readme-ov-file#scheduler) for detailed settings for `--optimizer_args` and `--lr_scheduler`.

**Comment by kohya-ss** [comment]
Prodigy can't be used with LoRA+ because LoRA+ uses a different learning rate for lora up. If you remove `--network_args loraplus_lr_ratio=4` it should work fine.

Edit: `--network_args loraplus_lr_ratio=0` also works, because the learning rate for lora_up is set to 0, this is fine for Prodigy.
